app/src/main/kotlin/com/mapzen/erasermap/view/MultiModalDirectionListAdapter.kt

```diff
@@ -211,7 +213,7 @@ class MultiModalDirectionListAdapter(val context: Context, val instructionGroupe
     }
     holder.instructionsContainer.removeAllViews()
     if (listItem.expanded) {
-      for (i in 0..instructionGroup.instructions.size - 1) {
+      for (i in instructionGroup.instructions.indices) {
```

Comment:
I was taking a look through the Kotlin reference last night and there appears to be the concept of enumerating items in a collection using a for loop (https://kotlinlang.org/docs/reference/control-flow.html#for-loops)

Would that be of any use in these scenarios? It seems like you're just pulling out the instructions.